device_services/generic/power_regulator.md

```diff
@@ -0,0 +1,110 @@
+# Power Regulator Service
+
+Power regulator service acts as a controller responsible for turning the relay on and off for a given part of the duty cycle.
+
+## Service name
+
+`power_regulator`
+
+## Interfaces
+
+| Type | Interface            | Value type | Description                    |
+|------|----------------------|------------|--------------------------------|
+| in   | cmd.lvl.set          | int        | Sets the level value.          |                                                                               
+| in   | cmd.lvl.get_report   | null       | Requests the level value.      |
+| out  | evt.lvl.report       | int        | Reports the level value.       |
+| -    |                      |            |                                |
+| in   | cmd.cycle.set        | int        | Sets the duty cycle value.     |                                                                               
+| in   | cmd.cycle.get_report | null       | Requests the duty cycle value. |
```

Comment:
What you call level here is in fact called duty cycle. Duty cycle is the ratio of time a load or circuit is ON compared to the time the load or circuit is OFF. Duty cycle, sometimes called "duty factor," is expressed as a percentage of ON time. A 60% duty cycle is a signal that is ON 60% of the time.

https://en.wikipedia.org/wiki/Duty_cycle

Reply:
You want to control 
'duty cycle' 
and
'period'

Reply:
Good point. I was translating in my head from polish "współczynnik" i "okres".